docs/streams/upgrade-guide.html

```diff
@@ -76,7 +76,14 @@ <h3><a id="streams_api_changes_240" href="#streams_api_changes_240">Streams API
 
     <!-- Placeholder KIP-213 -->
     <!-- Placeholder KIP-307 -->
-    <!-- Placeholder KIP-479 -->
+    <p>
+        Before the 2.4.0 version of Kafa Streams, users of the DSL could not name the state stores involved in a stream-stream join.  If users added DSL operations before of the 
+        join, the internal names of the state stores would shift, requiring an application reset when redeploying.  In the 2.4.0 release, Kafka Streams adds the <code>StreamJoined</code>
```

Comment:
`added DSL operations before of the join` -- sound confusing.

`If users changed their topology and added a operator before a join,`

Reply:
ack